pvlib/bifacial.py

```diff
@@ -177,13 +176,11 @@ def build(report, pvarray):
 
     @staticmethod
     def merge(reports):
-        """Works for dictionary reports"""
-        report = reports[0]
-        # Merge only if more than 1 report
-        if len(reports) > 1:
-            keys_report = list(reports[0].keys())
-            for other_report in reports[1:]:
-                if other_report is not None:
-                    for key in keys_report:
-                        report[key] += other_report[key]
+        """Works for dictionary reports. Merges the reports list of
+        dictionaries by flattening the lists for each key into a single
+        super list. Returns a dictionary with two list values."""
+        # Dictionary comprehension obviates the need to check if there are more
+        # than one report, and if one of the elements in reports is None.
+        report = {k: [item for d in reports for item in d[k]]
+                  for k in reports[0].keys()}
```

Comment:
I'm not at all familiar with `bifacial.py`. Is it clear that `reports[0].keys()` contains all the key values that any element of `reports` might have?

Reply:
That's  what the earlier implementation was doing anyway, so it's as sure as it was before. Each report seems to be coming from the build method just above the merge method, and if I'm correct about that, then each report dictionary in reports would have the same two keys.

Reply:
@anomam can you comment on this change? I want to make sure we're not missing something. If we are missing something, then we probably need a new test to catch it.

Reply:
Can you provide a test please? Also is it possible to use [`itertools.chain()`]? I don't think so, but maybe.
```python
>>> a = [{'a':[1,2],'b':[2,3],'c':[3,4]}, {'a':[44,55],'b':[55,66], 'c':[66,77]}, None, {'a':[777,888],'b':[888,999],'c':[999,1000]}]
>>> {k: [x for y in a for x in y[k]] for k in a[0].keys()}
a = [{'a':[1,2],'b':[2,3],'c':[3,4]}, {'a':[44,55],'b':[55,66], 'c':[66,77]}, {'a':[777,888],'b':[888,999],'c':[999,1000]}]
```
works, but I get `TypeError: 'NoneType' object is not subscriptable` if `None` is in `a`

Reply:
Since all reports are currently initialized with [this (line 162)](https://github.com/pvlib/pvlib-python/blob/e244343e97487113613005fa136c4964d27a2c03/pvlib/bifacial.py#L162)
```
report = {'total_inc_back': [], 'total_inc_front': []}
```
the ``NoneType`` error shouldn't be a problem. And it doesn't look to me like it's increasing the risk of having a bug compared to before. But I wouldn't be against a test for sure!
I agree that we lose in readability, but if the gain in performance is worth it, I think we should consider it. But it's up to the pvlib admins!
I don't think I will port this to the pvfactors package because of the readability concern, and because the reports in pvfactors are just there as examples; I use custom-made report classes in my work (and sometimes custom-made irradiance model classes) with pvfactors. Maybe that's something we could allow in pvlib as well?

Reply:
What is the gain in performance in this case?